example/kokkos_pi/kokkos_pi.cpp

```diff
@@ -0,0 +1,299 @@
+          if (lastBlock) {
+            // set pointers accordingly
+            lastBlock->next = pmb;
+            pmb->prev = lastBlock;
+          }
+
+          // store block away
+          lastBlock = allBlocks[idxMesh] = pmb;
+          // set coordinates of first cell center
+          h_xyz(0, idxMesh) = dxyzCell * (static_cast<Real>(i_mesh * n_block) + 0.5);
+          h_xyz(1, idxMesh) = dxyzCell * (static_cast<Real>(j_mesh * n_block) + 0.5);
+          h_xyz(2, idxMesh) = dxyzCell * (static_cast<Real>(k_mesh * n_block) + 0.5);
+          // Add our variable for in_or_out
+          Container<Real> &base = pmb->real_containers.Get();
+          base.setBlock(pmb);
+          base.Add("in_or_out", myMetadata);
+        }
+      }
+    }
+    // copy our coordinates over to Device and wait for completion
+    Kokkos::deep_copy(xyz, h_xyz);
+    Kokkos::fence();
+
+    // first let us do a stupid loop over the mesh
+    // This policy is over entire mesh
+
+    // This policy is over entire mesh
+    auto policy = Kokkos::RangePolicy<>(Kokkos::DefaultExecutionSpace(), 0,
+                                        n_block3 * n_mesh3, Kokkos::ChunkSize(512));
+
+    // This policy is over one block
+    auto policyBlock = Kokkos::RangePolicy<>(Kokkos::DefaultExecutionSpace(), 0, n_block3,
+                                             Kokkos::ChunkSize(512));
+
+    std::cout << "Begin basic timing " << std::endl;
+    double time_basic;
+    {
+      MeshBlock *pStart = allBlocks[0];
+      time_basic = kernel_timer_wrapper(0, n_iter, [&]() {
+        MeshBlock *pmb = pStart;
+        for (int iMesh = 0; iMesh < n_mesh3; iMesh++, pmb = pmb->next) {
+          Container<Real> &base = pmb->real_containers.Get();
+          auto inOrOut = base.PackVariables({Metadata::Independent});
+          // iops = 8  fops = 11
+          Kokkos::parallel_for(
+              "Compute In Or Out", policyBlock, KOKKOS_LAMBDA(const int &idx) {
+                const int k_grid = idx / n_block2;                             // iops = 1
+                const int j_grid = (idx - k_grid * n_block2) / n_block;        // iops = 3
+                const int i_grid = idx - k_grid * n_block2 - j_grid * n_block; // iops = 4
+                const Real x =
+                    xyz(0, iMesh) + dxyzCell * static_cast<Real>(i_grid); // fops = 2
+                const Real y =
+                    xyz(1, iMesh) + dxyzCell * static_cast<Real>(j_grid); // fops = 2
+                const Real z =
+                    xyz(2, iMesh) + dxyzCell * static_cast<Real>(k_grid); // fops = 2
+                const Real myR2 = x * x + y * y + z * z;                  // fops = 5
+                inOrOut(0, k_grid + NGHOST, j_grid + NGHOST, i_grid + NGHOST) =
+                    (myR2 < radius2 ? 1.0 : 0.0); // iops = 3
+              });
+        }
+      });
+      // wait for work to finish, redundant due to timing loop
+      Kokkos::fence();
```

Comment:
since this is redundant with the timing loop, I might remove it. It might introduce extra latency we can get rid of.